lib/redis_memo/middleware.rb

```diff
@@ -9,7 +9,9 @@ def call(env)
     result = nil
 
     RedisMemo::Cache.with_local_cache do
-      result = @app.call(env)
+      RedisMemo.with_max_connection_attempts(ENV['REDIS_MEMO_MAX_ATTEMPTS_PER_REQUEST']&.to_i) do
```

Comment:
I think at some point, we should also link the env vars with DefaultOptions. We can do that together along the other env vars later tho.